### PR TITLE
Trainee feature

### PR DESCRIPTION
**Pull request by fschueller**
included: 
- basic layout
- list of trainees with basic information
- homepage 
- create and edit functionality
- tests

please review

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -25,6 +25,10 @@ gem 'sdoc', '~> 0.4.0', group: :doc
 
 gem 'bootstrap-sass', '~> 3.3.5'
 
+gem 'haml'
```
Comment (kalabiyau) [inline Gemfile]: maybe remove?

File: app/controllers/trainees_controller.rb [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    @trainee = Trainee.find(params[:id])
+  end
+
+  def update
+    @trainee = Trainee.find(params[:id])
+    @trainee.update(trainee_params)
+    redirect_to(trainees_url)
+  end
+
+  def destroy
```
Comment (kalabiyau) [inline app/controllers/trainees_controller.rb]: maybe remove

File: app/models/trainee.rb [hunk 3]
```diff
@@ -0,0 +1,8 @@
+class Trainee < ActiveRecord::Base
+
+  validates :first_name, :last_name, :birthday, :year, :email, :phone, :suse_login, :github, :trello, :description, presence: :true
+
+  def full_name
```
Comment (kalabiyau) [inline app/models/trainee.rb]: please consider moving to decorator https://github.com/drapergem/draper

File: app/views/layouts/application.html.erb [hunk 4]
```diff
@@ -1,14 +1,71 @@
...
+          <div class="form-group">
+            <input type="text" class="form-control" placeholder="Password">
+          </div>
+          <button type="submit" class="btn btn-default">Login</button>
+        </form>
+      </div>
+    </div>
+  </nav>
+
+  <br>
```
Comment (kalabiyau) [inline app/views/layouts/application.html.erb]: better use css for that task

File: app/views/layouts/application.html.erb [hunk 5]
```diff
@@ -1,14 +1,71 @@
...
+
+  <footer class="footer navbar-fixed-bottom">
+    <div class="container">
+      <p class="text-muted">Made with love for SUSE's trainees. がんばろう!! </p>
+    </div>
+  </footer>
 
-<%= yield %>
 
+<script src="https://ajax.googleapis.com/ajax/libs/jquery/1.11.3/jquery.min.js"></script>
```
Comment (kalabiyau) [inline app/views/layouts/application.html.erb]: maybe use //ajax.. notation to fit the protocol of your project?

File: app/views/trainees/edit.html.erb [hunk 6]
```diff
@@ -0,0 +1,102 @@
+<div class="row">
+	<div class="col-md-3"></div>
+
+	<div class="col-md-6">
+
+		<div class="panel panel-success">
+			<div class="panel-heading">Edit your stuff!</div>
+
+		<div class="panel-body">
+			<%= form_for @trainee, :html => {:class => 'form-horizontal'}, url: trainee_url do |f| %>
```
Comment (kalabiyau) [inline app/views/trainees/edit.html.erb]: maybe try to use new ruby hash style?

File: app/models/trainee.rb [hunk 7]
```diff
@@ -0,0 +1,8 @@
+class Trainee < ActiveRecord::Base
+
+  validates :first_name, :last_name, :birthday, :year, :email, :phone, :suse_login, :github, :trello, :description, presence: :true
```
Comment (kalabiyau) [inline app/models/trainee.rb]: maybe we can add validations on email format and 
birthday?

File: db/migrate/20150812125119_create_trainees.rb [hunk 8]
```diff
@@ -0,0 +1,18 @@
+class CreateTrainees < ActiveRecord::Migration
+  def change
+    create_table :trainees do |t|
+      t.string :first_name
+      t.string :last_name
+      t.string :birthday
```
Comment (kalabiyau) [inline db/migrate/20150812125119_create_trainees.rb]: store with date?

File: spec/controllers/trainees_controller_spec.rb [hunk 9]
```diff
@@ -0,0 +1,39 @@
+require 'rails_helper'
+
+RSpec.describe TraineesController, type: :controller do
```
Comment (kalabiyau) [inline spec/controllers/trainees_controller_spec.rb]: i suggest to remove RSpec and type as those are rudimentary items

File: spec/controllers/trainees_controller_spec.rb [hunk 10]
```diff
@@ -0,0 +1,39 @@
...
+  let!(:trainee) { FactoryGirl.create(:trainee) }
```
Comment (kalabiyau) [inline spec/controllers/trainees_controller_spec.rb]: move to before block?

File: spec/controllers/trainees_controller_spec.rb [hunk 11]
```diff
@@ -0,0 +1,39 @@
...
+  let!(:params) { params = FactoryGirl.attributes_for(:trainee) }
```
Comment (kalabiyau) [inline spec/controllers/trainees_controller_spec.rb]: no assignment needed

Comment (kalabiyau) [inline spec/controllers/trainees_controller_spec.rb]: And maybe not ! version

File: spec/controllers/trainees_controller_spec.rb [hunk 12]
```diff
@@ -0,0 +1,39 @@
...
+
+  describe "GET #index" do
+    it "shows all trainees" do
+      trainees = Trainee.all
```
Comment (kalabiyau) [inline spec/controllers/trainees_controller_spec.rb]: we can move this to expect statement

File: spec/models/trainee_spec.rb [hunk 13]
```diff
@@ -0,0 +1,39 @@
+require 'rails_helper'
+
+RSpec.describe Trainee, type: :model do
```
Comment (kalabiyau) [inline spec/models/trainee_spec.rb]: Maybe use https://github.com/thoughtbot/shoulda-matchers here - as it is more readable

File: spec/models/trainee_spec.rb [hunk 14]
```diff
@@ -0,0 +1,39 @@
...
+      expect(trainee).to be_valid
+    end
+
+    it "should concatenate first and last name" do
+      trainee = Trainee.new(first_name: "Farah", last_name: "Rabea")
+      expect(trainee.full_name).to eq 'Farah Rabea'
+    end
+
+  end
+end
```
Comment (kalabiyau) [inline spec/models/trainee_spec.rb]: newline ?

File: app/views/layouts/application.html.erb [hunk 15]
```diff
@@ -1,14 +1,66 @@
...
+          <span class="icon-bar"></span>
+        </button>
+        <a class="navbar-brand" href="#">SUSE Trainee Dashboard</a>
+      </div>
+    	<div id="navbar" class="collapse navbar-collapse">
+    		<ul class="nav navbar-nav">
+          <li class="active"><a href="/">Home</a></li>
+            <li class="dropdown">
+              <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-haspopup="true" aria-expanded="false">Trainees <span class="caret"></span></a>
+              <ul class="dropdown-menu">
```
Comment (NadjaW) [inline app/views/layouts/application.html.erb]: please remove this dropdown from header of the page and move it to the left of the page. 
create a menu list like on teamdashboard

File: app/views/trainees/index.html.erb [hunk 16]
```diff
@@ -0,0 +1,51 @@
+<div class="row">
+	<div class="col-md-6">
+	<div class="panel panel-default">
+		<div class="panel-body">
+			<h2>The trainee elite of SUSE :D</h2>
+			<p>
```
Comment (NadjaW) [inline app/views/trainees/index.html.erb]: please change headline to SUSE Trainees 1st year of apprenticeship, 2nd year... depends on which menu entry you clicked on the left menu

File: app/views/layouts/application.html.erb [hunk 17]
```diff
@@ -1,14 +1,66 @@
...
+                <li><a href="/trainees">1st year</a></li>
+                <li><a href="/trainees">2nd year</a></li>
+                <li><a href="/trainees">3rd year</a></li>
+              </ul>
+            </li>
+          <li><a href="#contact">Contact</a></li>
```
Comment (NadjaW) [inline app/views/layouts/application.html.erb]: contact link doesnt work and i didnt think about if i want to have a contact page. please remove it for now, i will add a card if needed

File: app/views/layouts/application.html.erb [hunk 18]
```diff
@@ -1,14 +1,66 @@
...
+        </ul>
+        <form class="navbar-form navbar-right" role="login">
```
Comment (NadjaW) [inline app/views/layouts/application.html.erb]: can you please remove login fields, they were not part of the card. login is not needed for the upcoming information on the page

File: app/views/trainees/edit.html.erb [hunk 19]
```diff
@@ -0,0 +1,102 @@
...
+      <div class="form-group">
+        <%= f.label :trello, :class => "control-label col-md-2" %>
+        <div class="col-md-7">
+          <%= f.text_field :trello, :class => "form-control", required: true %>
+        </div>
+        <div class="col-md-offset-3"></div>
+      </div>
+
+      <div class="form-group">
+        <%= f.label :description, :class => "control-label col-md-2" %>
```
Comment (NadjaW) [inline app/views/trainees/edit.html.erb]: please add more information to description field. 
in trello card it is saying: small description about what he/she did before SUSE, i guess thats not clear to everyone if we only say: description ;)

File: app/views/trainees/new.html.erb [hunk 20]
```diff
@@ -0,0 +1,108 @@
...
+      <div class="form-group">
+        <%= f.label :last_name, :class => "control-label col-md-2" %>
+        <div class="col-md-7">
+          <%= f.text_field :last_name, :class => "form-control", required: true %>
+        </div>
+        <div class="col-md-offset-3"></div>
+      </div>
+
+      <div class="form-group">
+        <%= f.label :birthday, :class => "control-label col-md-2" %>
```
Comment (NadjaW) [inline app/views/trainees/new.html.erb]: please add a birthday field here like on teamdashboard. it provides a small calender, where you can click on a date :)

File: app/assets/stylesheets/application.scss [hunk 21]
```diff
@@ -10,6 +10,14 @@
...
+ *= require_self
  */
  @import "bootstrap-sprockets";
  @import "bootstrap";
+
+.container-fluid {
+
+	padding-top: 70px;
+
+}
```
Comment (kalabiyau) [inline app/assets/stylesheets/application.scss]: newline?

File: app/assets/stylesheets/application.scss [hunk 22]
```diff
@@ -10,6 +10,14 @@
...
  *
+ *= require_tree
+ *= require_self
  */
  @import "bootstrap-sprockets";
  @import "bootstrap";
+
+.container-fluid {
+
+	padding-top: 70px;
```
Comment (kalabiyau) [inline app/assets/stylesheets/application.scss]: wrong indentation?

File: app/views/dashboard/index.html.erb [hunk 23]
```diff
@@ -1,2 +1,4 @@
-<h1>Dashboard#index</h1>
-<p>Find me in app/views/dashboard/index.html.erb</p>
+<div class="jumbotron">
+	<h1>Welcome to the SUSE Trainee Dashboard!</h1>
```
Comment (kalabiyau) [inline app/views/dashboard/index.html.erb]: broken indentation?

File: app/views/layouts/application.html.erb [hunk 24]
```diff
@@ -1,14 +1,66 @@
 <!DOCTYPE html>
 <html>
 <head>
+	<meta charset="utf-8">
```
Comment (kalabiyau) [inline app/views/layouts/application.html.erb]: broken indentation

File: app/views/trainees/index.html.erb (hunk 16 above)
Comment (fschueller) [inline app/views/trainees/index.html.erb]: This feature is included in another recent feature branch i developed on top of the trainee_feature branch while the pull request was being reviewed. I still have to sort out the merge issues.

File: app/views/trainees/edit.html.erb (hunk 19 above)
Comment (fschueller) [inline app/views/trainees/edit.html.erb]: Added in edit.html.erb and new.html.erb (see commits below)

File: app/views/trainees/new.html.erb (hunk 20 above)
Comment (fschueller) [inline app/views/trainees/new.html.erb]: I'd like to add this in a new feature branch if possible

**Pull request merged**